.changeset/weak-mayflies-search.md

```diff
@@ -0,0 +1,21 @@
+---
+'@compiled/parcel-transformer': patch
+'@compiled/webpack-loader': patch
+'@compiled/babel-plugin': patch
+'@compiled/utils': patch
+'@compiled/css': patch
+---
+
+Add the `enforcePseudoOrder` option. This ensures that the pseudo-selectors from Compiled are always applied in the order
+
+- `:link`
+- `:visited`
+- `:focus-within`
+- `:focus`
+- `:focus-visible`
+- `:hover`
+- `:active`
+
+even when components are imported from multiple packages. This is at the expense of lengthening those selectors considerably (by adding `:not(#\\##\\##\\##\\##\\#)` to the end of those selectors).
```

Comment:
…following our conversation today @JakeLane @itsdouges @dddlr, does this ever result in the same issue we had with mixing specificity as with `increaseSpecificity: true`?  It feels like the answer is yes, and this is a very late idea, but is this safe in the internal scenario we identified?

Reply:
yes

Reply:
if someone mixes Compiled with their own CSS solution :/